airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py

```diff
@@ -148,7 +150,17 @@ def get_field_value(field_value: GoogleAdsRow, field: str) -> str:
         return field_value
 
     @staticmethod
-    def parse_single_result(schema: Mapping[str, Any], result: GoogleAdsRow):
-        fields = GoogleAds.get_fields_from_schema(schema)
+    def process_query(query) -> List:
+        query = query.lower().split("select")[1].split("from")[0].strip()
+        fields = query.split(",")
+        fields = [i.strip() for i in fields]
+        return fields
+
+    @staticmethod
+    def parse_single_result(schema: Mapping[str, Any], result: GoogleAdsRow, query: str = None):
+        if not query:
+            fields = GoogleAds.get_fields_from_schema(schema)
+        else:
+            fields = GoogleAds.process_query(query)
```

Comment:
this should be part of the schema, i.e. `get_json_schema` should return correct schema